python/ray/data/expressions.py

```diff
@@ -0,0 +1,317 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any
+
+from ray.util.annotations import PublicAPI
+
+
+@PublicAPI(stability="alpha")
+class Operation(Enum):
```

Comment:
This is not a public API (only stuff that users of RD will be using is -- Dataset APIs, etc)